src/frame.zig

```diff
@@ -197,86 +197,77 @@ pub fn init(a: Allocator, req: *const Request, auth: Auth.Provider) !Frame {
     };
 }
 
+fn VecList(comptime SIZE: usize) type {
+    return struct {
+        pub const capacity = SIZE;
+        vect: [SIZE]iovec_c = undefined,
+        length: usize = 0,
+
+        pub fn init() @This() {
+            return .{};
+        }
+
+        pub fn append(self: *@This(), str: []const u8) !void {
+            if (self.length >= capacity) return error.NoSpaceLeft;
+            self.vect[self.length] = .{
+                .base = str.ptr,
+                .len = str.len,
+            };
+            self.length += 1;
+        }
```

Comment:
If you agree with the changes to Cookie and cookie attribute structs then a function could be added to append a `[]iovec_const`